src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzer.cs

```diff
@@ -0,0 +1,446 @@
+                {
+                    runtimeInformationType = context.Compilation.GetOrCreateTypeByMetadataName(typeName);
+                }
+                if (!context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemRuntimeInteropServicesOSPlatform, out var osPlatformType))
+                {
+                    return;
+                }
+
+                var guardMethods = GetRuntimePlatformGuardMethods(runtimeInformationType!);
+
+                context.RegisterOperationBlockStartAction(context => AnalyzerOperationBlock(context, guardMethods, osPlatformType));
+            });
+
+            static ImmutableArray<IMethodSymbol> GetRuntimePlatformGuardMethods(INamedTypeSymbol runtimeInformationType)
+            {
+                return runtimeInformationType.GetMembers().OfType<IMethodSymbol>().Where(m =>
+                    s_platformCheckMethodNames.Contains(m.Name) && !m.Parameters.IsEmpty).ToImmutableArray();
+            }
+        }
+
+        private void AnalyzerOperationBlock(OperationBlockStartAnalysisContext context, ImmutableArray<IMethodSymbol> guardMethods, INamedTypeSymbol osPlatformType)
+        {
+            var platformSpecificOperations = PooledDictionary<IOperation, ImmutableArray<PlatformAttributeInfo>>.GetInstance();
+
+            context.RegisterOperationAction(context =>
+            {
+                AnalyzeInvocationOperation(context.Operation, context, ref platformSpecificOperations);
+            }
+            , OperationKind.DelegateCreation, OperationKind.EventReference, OperationKind.FieldReference,
+            OperationKind.Invocation, OperationKind.ObjectCreation, OperationKind.PropertyReference);
+
+            context.RegisterOperationBlockEndAction(context =>
+            {
+                try
+                {
+                    if (platformSpecificOperations.Count == 0)
+                    {
+                        return;
+                    }
+
+                    if (guardMethods.IsEmpty || !(context.OperationBlocks.GetControlFlowGraph() is { } cfg))
+                    {
+                        ReportDiagnosticsForAll(platformSpecificOperations, context);
+                        return;
+                    }
+
+                    var wellKnownTypeProvider = WellKnownTypeProvider.GetOrCreate(context.Compilation);
+                    var analysisResult = GlobalFlowStateAnalysis.TryGetOrComputeResult(
+                        cfg, context.OwningSymbol, CreateOperationVisitor, wellKnownTypeProvider,
+                        context.Options, MinimumOsRule, performValueContentAnalysis: true,
+                        context.CancellationToken, out var valueContentAnalysisResult);
+
+                    if (analysisResult == null)
+                    {
+                        ReportDiagnosticsForAll(platformSpecificOperations, context);
```

Comment:
This seems too aggressive and likely leads to false positives. The fact that flow analysis did not succeed should mean we report all, instead we should bail out from reporting any.

Reply:
It because the overall evaluation not only depends on flow analysis, even it was not successful other evaluations would still suggest warning. But I agree it could lead to false positive, will remove.